### ISSUE TITLE
0.7.1 Lost leading "/" from "/var/cache/renderd/tiles"

### ISSUE THREAD

**Issue by linsomniac** [openstreetmap/mod_tile#416]
0.7.0 mod_tile has a RENDERD_TILE_DIR set in `CMakeLists.txt`:

    set(RENDERD_TILE_DIR "/${CMAKE_INSTALL_LOCALSTATEDIR}/cache/renderd/tiles")

and in 0.7.1 it was changed to:

    set(RENDERD_TILE_DIR "${CMAKE_INSTALL_LOCALSTATEDIR}/cache/renderd/tiles")

Note the lack of the leading slash.

We have a default install of mod_tile that we've been using, and when it switched to not have the leading slash, it caused our setup to fail because it was trying to access a relative directory, I'm assuming FQN of `/etc/apache2/var/cache/renderd/tiles`

I'm assuming one fix is for us to set up an apache2 conf file with "ModTileTileDir /var/cache/renderd/tiles" in it, but:

- Probably should be noted that the default has been changed, as no mention of this was made in the 0.7.1 release notes.
- Would be interested in knowing why this was changed to drop the leading "/", because
- Perhaps this will have other impacts as well?

**Comment by linsomniac** [comment]
In further looking, WRT `docs/build/building_on_ubuntu.md`, it looks like one fix will be in our building script set, among other things, "CMAKE_INSTALL_LOCALSTATEDIR=/var".

Again, going to propose that the (apparent) change of "CMAKE_INSTALL_LOCALSTATEDIR" leading "/" be mentioned in changelog.

**Comment by hummeltech** [comment]
Are you building this yourself, or are you using the newly released `Debian Unstable` package? The change you're mentioning was made at the request of the Debian package maintainer, and as per CMake recommended usage. Using absolute paths when setting install target paths, such as `RENDERD_TILE_DIR`, causes issues when specifying custom `CMAKE_INSTALL_*` paths during a package build using `DESTDIR`.

If you are using the `Debian Unstable` package, the default value for `RENDERD_TILE_DIR` is absolute (which is generated from `CMAKE_INSTALL_LOCALSTATEDIR` + `/cache/renderd/tiles` => `/var/cache/renderd/tiles`.)

However if you are building `mod_tile` yourself and not specifying an absolute path for `CMAKE_INSTALL_LOCALSTATEDIR` as updated in the documentation then it will default to `CMAKE_INSTALL_PREFIX` + `CMAKE_LOCALSTATEDIR` + `/cache/renderd/tiles` which will yield `/usr/local/var/cache/renderd/tiles`.

You can check what default value is compiled into your binary using `strings`:
```shell
$ strings /usr/lib/apache2/modules/mod_tile.so | grep cache/renderd/tiles
/var/cache/renderd/tiles
```

**Comment by hummeltech** [comment]
I have updated the release notes:
https://github.com/openstreetmap/mod_tile/releases/tag/v0.7.1

Let me know if that is not what you had in mind.

Thanks again for the report.

**Comment by linsomniac** [comment]
Thanks for the response, and just to clarify I think we have the resolution we will need going forward (setting the CMAKE_INSTALL values).  Your changes to the changelog look good, thank you!

You mention that the default, if self-building, will be "/usr/local/var/cache/renderd/tiles", but our experience, and I'm not seeing that we were specifying anything to override it, was "var/cache/renderd/tiles" (without leading "/").  I'm not exactly sure where our build recipe came from, a coworker set that up.  I'm going to refine it to bring it up to the current set of "building on Ubuntu" instructions.  We are building on Ubuntu 23.10 at the moment, as we needed newer mapnik built with gdal which was not available in 22.04.

I'm going to close this with this comment as it's fully resolved from my standpoint, I'm providing additional information above simply for completeness and help if someone else runs across issues and can benefit from it.
